ntbs-bulk-insert/Program.cs

```diff
@@ -32,46 +33,84 @@ static async Task Main(string[] args)
         {
-            var numberOfNotifications = 30000;
+            var numberOfNotifications = 3000;
+            var rand = new Random();
+            var tbServices = (await context.TbService.ToListAsync()).Select(t => t.Code).ToList();
             var notificationsOperable = Builder<Notification>.CreateListOfSize(numberOfNotifications)
                 .All()
                 .With(n => n.NotificationId = 0)
                 .With(n => n.NotificationStatus = NotificationStatus.Notified)
                 .With(n => n.PatientDetails.GivenName = Faker.Name.First())
                 .With(n => n.PatientDetails.FamilyName = Faker.Name.Last())
-                .With(n => n.PatientDetails.NhsNumberNotKnown = true)
-                .With(n => n.HospitalDetails.TBServiceCode = "TBS0008")
-                .With(n => n.HospitalDetails.Consultant = "UniqueBulkInsert")
-                .With(n => n.NotificationDate = new DateTime(2017, 1, 1))
-                .With(n => n.GroupId = null);
-
-            // Comment out line below to not add the Treatment event which trigger Treatment Outcome DQ alerts
-            notificationsOperable = AddDataQualityTreatmentEvents(notificationsOperable);
+                .With(n => n.ClinicalDetails.Notes = "UniqueBulkInsert")
+                .With(n => n.GroupId = null)
+                // Add randomised fields that Faker cannot generate
+                .With(n => n.NotificationDate = AddRandomDateTimeBetween2014And2017(rand))
+                .With(n => n.HospitalDetails.TBServiceCode = AddRandomTbService(rand, tbServices))
+                .With(n => n.PatientDetails.NhsNumber = AddRandomTestNhsNumber(rand));
 
+            // Comment out the line below to cause Treatment Outcome Alerts to be generated
+            // notificationsOperable = await AddDataQualityTreatmentEvents(notificationsOperable, context);
+            
             var notifications= notificationsOperable.Build();
+
             context.AddRange(notifications);
             await context.SaveChangesAsync();
         }
 
-        private static IOperable<Notification> AddDataQualityTreatmentEvents(IOperable<Notification> notificationsOperable)
+        private static DateTime AddRandomDateTimeBetween2014And2017(Random rand)
         {
+            var days = rand.Next(1, 1000);
+            return new DateTime(2014, 1, 1).AddDays(days);
+        }
+
+        private static string AddRandomTbService(Random rand, IList<string> tbServices)
+        {
+            var tbServiceIndex = rand.Next(0, tbServices.Count - 1);
+            return tbServices[tbServiceIndex];
+        }
+
+        private static string AddRandomTestNhsNumber(Random rand)
+        {
+            var nhsNumberString = "9";
+            for (var i = 0; i < 9; i++)
+            {
+                nhsNumberString = String.Concat(nhsNumberString, rand.Next(1, 9).ToString());
+            }
+            return nhsNumberString;
```

Comment:
I wouldn't know whether it's a worth-while saving without running it, but this seems like exactly the type of code that wants to use `StringBuilder` over a `string`. Given the idea is to have this code run tens of thousands of times, it might just be noticeable to generate a single string, rather than 10 of them.